cardano-explorer-node/src/Explorer/Node/Database.hs

```diff
@@ -0,0 +1,111 @@
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Explorer.Node.Database
+  ( DbAction (..)
+  , DbActionQueue (..)
+  , lengthDbActionQueue
+  , newDbActionQueue
+  , runDbThread
+  , writeDbActionQueue
+  ) where
+
+import           Cardano.BM.Trace (Trace, logDebug, logInfo)
+import           Cardano.Prelude
+
+import qualified Control.Concurrent.STM as STM
+import           Control.Concurrent.STM.TBQueue (TBQueue)
+import qualified Control.Concurrent.STM.TBQueue as TBQ
+
+import qualified Explorer.DB as DB
+import           Explorer.Node.Insert
+import           Explorer.Node.Metrics
+import           Explorer.Node.Rollback
+import           Explorer.Node.Util
+
+import           Ouroboros.Consensus.Ledger.Byron (ByronBlockOrEBB (..))
+import           Ouroboros.Consensus.Ledger.Byron.Config (ByronConfig)
+import           Ouroboros.Network.Block (BlockNo (..), Point (..))
+
+import qualified System.Metrics.Prometheus.Metric.Gauge as Gauge
+
+
+data NextState
+  = Continue
+  | Done
+
+data DbAction
+  = DbApplyBlock !(ByronBlockOrEBB ByronConfig) !BlockNo
+  | DbRollBackToPoint !(Point (ByronBlockOrEBB ByronConfig))
+  | DbFinish
+
+newtype DbActionQueue = DbActionQueue
+  { dbActQueue :: TBQueue DbAction
+  }
+
+lengthDbActionQueue :: DbActionQueue -> STM Natural
+lengthDbActionQueue (DbActionQueue q) = STM.lengthTBQueue q
+
+newDbActionQueue :: IO DbActionQueue
+newDbActionQueue = DbActionQueue <$> TBQ.newTBQueueIO 2000
+
+writeDbActionQueue :: DbActionQueue -> DbAction -> STM ()
+writeDbActionQueue (DbActionQueue q) = TBQ.writeTBQueue q
+
+
+runDbThread :: Trace IO Text -> Metrics -> DbActionQueue -> IO ()
+runDbThread trce metrics queue = do
+    logInfo trce "Running DB thread"
+    loop
+    mBlkNo <-  DB.runDbNoLogging DB.queryLatestBlockNo
+    case mBlkNo of
+      Nothing -> pure ()
+      Just blkNo -> Gauge.set (fromIntegral blkNo) $ mDbHeight metrics
```

Comment:
`mDbHeight` is only updated once, as the db thread shuts down, and never gets updated after inserting things

Reply:
Fixing that!.